precompiles/outposts/osmosis/tx.go

```diff
@@ -59,26 +59,40 @@ func (p Precompile) Swap(
 		return nil, err
 	}
 
-	// We need to check if the input and output denom exist. If they exist we retrieve their denom
-	// otherwise error out.
-	inputDenom, err := p.erc20Keeper.GetTokenDenom(ctx, input)
-	if err != nil {
-		return nil, err
+	bondDenom := p.stakingKeeper.GetParams(ctx).BondDenom
+	var inputDenom, outputDenom string
```

Comment:
Should we check that the output != input before looking up for the denom?
We can save a couple of DB reads

Reply:
It would screw up the tests and flow because the check are done after the correct input and output denom are taken from the tokenPair or populated with the base denom.